larky/src/main/java/com/verygood/security/larky/modules/vgs/cerebro/text/analyzer/dto/TextPIIEntity.java

```diff
@@ -69,16 +72,24 @@ public StarlarkInt end() {
     return end;
   }
 
-  @StarlarkMethod(
-      name = "to_dict",
-      doc = "Dictionary PII representation")
-  public Dict<String, Object> toDict() {
+  @Override
+  @StarlarkMethod(name = PyProtocols.__DICT__, structField = true)
+  public Dict<String, Object> dunderDict() {
+    return TextPIIEntity.dictOf(this.entityType, this.score, this.start, this.end);
+  }
+
+  @Override
+  public boolean isImmutable() {
+    return true;
+  }
+
+  private static Dict<String, Object> dictOf(String entityType, StarlarkFloat score, StarlarkInt start,
+                                             StarlarkInt end) {
     return Dict.<String, Object>builder()
-        .put("entity_type", this.entityType)
-        .put("score", this.score)
-        .put("start", this.start)
-        .put("end", this.end)
+        .put("entity_type", entityType)
```

Comment:
do you want to sort the keys to make this deterministic?

Reply:
Updated